### PR TITLE
Show merchant names on card grant pages

### PR DESCRIPTION
**Pull request by 24c02**
## Summary of the problem
people receiving card grants can't see what merchants their grants can be spent on, which will cause confusion for those with multiple grants for different purposes.

## Describe your changes
i added an "allowed merchants" section to the spending instructions view, which shows merchant names from yellow_pages, or failing that just the raw MIDs.

while i was in there, i also made the organizer view of card grants show friendly names instead of raw MIDs :-)

organizer view:
Screenshot 2024-10-02 at 12 31 39 PM
grantee view:
Screenshot 2024-10-02 at 12 32 05 PM

### CODE REVIEW

File: app/controllers/card_grants_controller.rb [hunk 1]
```diff
@@ -60,6 +60,7 @@ def show
 
     @frame = params[:frame].present?
     @force_no_popover = @frame
+    @allowed_merchants = @card_grant.allowed_merchants.map {|merchant_id| YellowPages::Merchant.lookup(network_id: merchant_id).name || merchant_id}.to_set
```
Comment (garyhtou) [inline app/controllers/card_grants_controller.rb]: I'd recommend moving this logic into the model under a method called `allowed_merchant_names`. Then, you can call that method directly from the view!

**Comment by garyhtou** [comment]
Thank you! I love how you're helping improve the card grant experience for everyone

File: app/controllers/card_grants_controller.rb (hunk 1 above)
Comment (24c02) [reply app/controllers/card_grants_controller.rb]: done!

**Comment by 24c02** [comment]
## newest visual changes:
organizer/admin:
Screenshot 2024-10-02 at 4 45 40 PM
grantee:
Screenshot 2024-10-02 at 4 46 45 PM

**Review by garyhtou — DISMISSED** [review]
Thank you for the screenshots! I'm happy with this. Can you run `bin/lint` to fix up rubocop/erblint?

**Review by sampoder — APPROVED** [review]
LGTM; thanks so much Nora!

**Pull request merged**